pkg/controller/networkpolicyrecommendation/controller.go

```diff
@@ -118,19 +205,64 @@ func (c *NPRecommendationController) Run(stopCh <-chan struct{}) {
 		return
 	}
 
+	go func() {
+		wait.Until(c.resyncNPRecommendation, npRecommendationResyncPeriod, stopCh)
+	}()
+
+	go wait.Until(c.deletionworker, time.Second, stopCh)
+
 	for i := 0; i < defaultWorkers; i++ {
 		go wait.Until(c.worker, time.Second, stopCh)
 	}
 	<-stopCh
 }
 
+func (c *NPRecommendationController) deletionworker() {
```

Comment:
We need some GC logic to recycle stale resources (rnp and db entry) in case npr is removed during manager downtime, or manager is restarted without finishing the cleanup job. this may be addressed in a future pr.

Reply:
The idea of GC sounds great to me. As we may remove rnp in next release, maybe it will be a periodic job checking db to remove an entry with an id not in the current npr list?
Besides, I'm also thinking about whether to do the similar GC for the stale SparkApplication. I'm a little bit worry about whether this will delete some jobs not created by the manager in the future.

Reply:
one way around is we introduce additional metadata (annotation / labels) for the ones created by Theia. This way we won't bother with other unrelated SparkApplications. Periodical is good but at least this should be triggered once per mgr init.